openpype/hosts/blender/plugins/publish/validate_mesh_has_uv.py

```diff
@@ -33,20 +34,21 @@ def has_uvs(obj: bpy.types.Object) -> bool:
     @classmethod
     def get_invalid(cls, instance) -> List:
         invalid = []
-        # TODO (jasper): only check objects in the collection that will be published?
-        for obj in [
-            obj for obj in instance]:
+        for obj in set(instance):
             try:
-                if obj.type == 'MESH':
-                    # Make sure we are in object mode.
-                    bpy.ops.object.mode_set(mode='OBJECT')
+                if isinstance(obj, bpy.types.Object) and obj.type == 'MESH':
+                    if obj.mode != 'OBJECT':
+                        # Make sure we are in object mode.
+                        bpy.ops.object.mode_set(mode='OBJECT')
```

Comment:
Somehow I feel this check should be outside the for loop since it looks like we're setting the object mode globally? Couldn't we query that globally as well? (This isn't necessarily a comment about the code change - it was there originally too)

Reply:
Yes absolutely and I'm not happy with this ops and I think it should stay as it to be consistent with the Object mode Validator.